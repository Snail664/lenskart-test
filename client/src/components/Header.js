import React from "react";
import { Link } from "react-router-dom";

const Header = ({ prevPage }) => {
  return (
    <div>
      <Link to={`${prevPage}`}>Back</Link>
    </div>
  );
};

export default Header;
