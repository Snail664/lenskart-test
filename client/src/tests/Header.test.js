import React from "react";
import "@testing-library/jest-dom/extend-expect";
import ShallowRenderer from "react-test-renderer/shallow";
import { Router } from "react-router-dom";
import history from "../history";

import Header from "../components/Header";

// header renders correctly
it("Matches snapshot for loaded props", () => {
  const renderer = new ShallowRenderer();
  renderer.render(
    <Router history={history}>
      <Header prevPage="/test" />
    </Router>
  );
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
