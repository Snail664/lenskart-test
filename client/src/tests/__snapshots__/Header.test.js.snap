// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot for loaded props 1`] = `
<Context.Provider
  value={
    Object {
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "match": Object {
        "isExact": true,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <Context.Provider
    value={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header
      prevPage="/test"
    />
  </Context.Provider>
</Context.Provider>
`;
